packages/build/src/plugins_core/functions/zisi.js

```diff
@@ -0,0 +1,54 @@
+import { join, resolve } from 'path'
+
+import mapObject from 'map-obj'
+
+import { getZisiFeatureFlags } from './feature_flags.js'
+
+export const getZisiParameters = ({
+  buildDir,
+  childEnv,
+  featureFlags,
+  functionsConfig,
+  functionsDist,
+  isRunningLocally,
+  repositoryRoot,
+}) => {
+  const nodeVersion = childEnv.AWS_LAMBDA_JS_RUNTIME
+  const isManifestEnabled = isRunningLocally || featureFlags.buildbot_create_functions_manifest === true
+  const manifest = isManifestEnabled ? join(functionsDist, 'manifest.json') : undefined
+  const config = mapObject(functionsConfig, (expression, object) => [
+    expression,
+    normalizeFunctionConfig({ buildDir, featureFlags, functionConfig: object, isRunningLocally, nodeVersion }),
+  ])
+  const zisiFeatureFlags = getZisiFeatureFlags(featureFlags)
+
+  return { basePath: buildDir, config, manifest, featureFlags: zisiFeatureFlags, repositoryRoot }
+}
+
+// The function configuration keys returned by @netlify/config are not an exact
+// match to the properties that @netlify/zip-it-and-ship-it expects. We do that
+// translation here.
+export const normalizeFunctionConfig = ({ buildDir, functionConfig = {}, isRunningLocally, nodeVersion }) => ({
+  externalNodeModules: functionConfig.external_node_modules,
+  includedFiles: functionConfig.included_files,
+  includedFilesBasePath: buildDir,
+  ignoredNodeModules: functionConfig.ignored_node_modules,
+  nodeVersion,
```

Comment:
Just to confirm: does ZISI handle correctly `nodeVersion` not being one of the acceptable values? For example, if the user mistyped it. Or if the value is an empty string. And so on.

Reply:
Nice!